doc/migrations/v0.29-v0.30.md

```diff
@@ -0,0 +1,191 @@
+
+The signing property is now based on a `globalSignaturePolicy` option instead of the previous `signMessages` and `strictSigning` options. The default to strict signing pubsub messages was kept, but if you would like to disable it, the properties should be changed as follows:
+
+**Before**
+
+```js
+const Gossipsub = require('libp2p-gossipsub')
+const Libp2p = require('libp2p')
+
+const libp2p = await Libp2p.create({
+  modules: {
+    // ... Add required modules according to the Configuration docs
+    pubsub: Gossipsub
+  },
+  config: {
+    pubsub: {
+      signMessages: false,
+      strictSigning: false
+    }
+  }
+})
+```
+
+**After**
+
+```js
+const Gossipsub = require('libp2p-gossipsub')
+const { SignaturePolicy } = require('libp2p-interfaces/src/pubsub/signature-policy')
+const Libp2p = require('libp2p')
+
+const libp2p = await Libp2p.create({
+  modules: {
+    // ... Add required modules according to the Configuration docs
+    pubsub: Gossipsub
+  },
+  config: {
+    pubsub: {
+      globalSignaturePolicy: SignaturePolicy.StrictNoSign
+    }
+  }
+})
+```
+
+### Addresses
+
+Libp2p has supported `noAnnounce` addresses configuration for some time now. However, it did not provide the best developer experience. In this release, we dropped the `noAnnounce` configuration property in favor of an `announceFilter` property function.
+
+**Before**
+
+```js
+const Libp2p = require('libp2p')
+
+// ... Create/Load PeerId
+
+const libp2p = await Libp2p.create({
+  modules: {
+    // ... Add required modules according to the Configuration docs
+  },
+  peerId,
+  addresses: {
+    listen: ['/ip4/127.0.0.1/tcp/8000/ws'],
+    noAnnounce: ['/ip4/127.0.0.1/tcp/8000/ws'],
+  }
+})
+```
+
+**After**
+
+```js
+const Libp2p = require('libp2p')
+
+// Libp2p utils has several multiaddr utils you can leverage
+const isPrivate = require('libp2p-utils/src/multiaddr/is-private')
+
+// ... Create/Load PeerId
+
+const libp2p = await Libp2p.create({
+  modules: {
+    // ... Add required modules according to the Configuration docs
+  },
+  peerId,
+  addresses: {
+    listen: ['/ip4/127.0.0.1/tcp/8000/ws'],
+    // Filter function: (ma: Array<multiaddr>) => Array<multiaddr>
+    announceFilter: (multiaddrs) => multiaddrs.filter(m => !isPrivate(m))
+  }
+})
+```
+
+It is important pointing out another change regarding address advertising. This is not an API breaking change, but it might have influence on your libp2p setup.
+Previously, when using the addresses `announce` property, its multiaddrs were concatenated with the `listen` multiaddrs and then they were filtered out by the `noAnnounce` multiaddrs, in order to create the list of multiaddrs to advertise. 
+In `libp2p@0.30`, we are taking the same approach as `go-libp2p` and if `announce` addresses are provided, only them are used for advertising. If no addresses are provided, the advertised addresses will be the `listen` addresses filtered out by the `announceFilter` (if provided).
```

Comment:
I think it might be helpful to also reword this section to itemize it for clarity:
```suggestion
In `libp2p@0.30` the logic now operates as follows:

- If `announce` addresses are provided, only they will be announced (no filters are applied)
- If `announce` is not provided, the transport addresses will be filtered (if a filter is provided)
  - if the `announceFilter` is provide it will be passed the transport addresses
```